src/product/product.service.ts

```diff
@@ -28,6 +29,18 @@ export class ProductService implements IProductService {
   async createProduct(product: CreateProductDTO): Promise<ProductResponseDto> {
     const newProduct = await this.productRepository.save(product);
     return ProductResponseDto.mapToResponse(newProduct);
+  }
 
+  async deleteProduct(id: string): Promise<Boolean>{
+      if (await this.productRepository.exist({where: {id}})) {
+          this.productRepository.update(id, {isActive:false});
+          return true
+      }else {
+        return false
+      }
   }
```

Comment:
async deleteProduct(id: string): Promise<void>{
    const exists = await this.productRepository.exist({where: {id}})
    if (!exist) {
      throw new BadRequestException('Product not found');
     }     
   await this.productRepository.update(id, {isActive:false});
   } // It's not necessary return something